lms/djangoapps/grades/api/views.py

```diff
@@ -128,10 +137,18 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
-        username = request.GET.get('username')
 
-        # only the student can access her own grade status info
-        if request.user.username != username:
+        # Use the request user's if none provided.
+
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
+
+        def user_mismatch_error():
+            """
+            Log the attempt, and return a 404, user_mismatch response.
```

Comment:
I concur!  Fixed here: https://github.com/edx/edx-platform/pull/14710/commits/eb11c98ce6d52cb84b3f02b79a0b3e71b9224a63#diff-dd3ce32b0bf0e8e67e0d10618ea0d4f7R147

Have also added an [additional error handler](https://github.com/edx/edx-platform/pull/14710/commits/eb11c98ce6d52cb84b3f02b79a0b3e71b9224a63#diff-dd3ce32b0bf0e8e67e0d10618ea0d4f7R169) and [test case](https://github.com/edx/edx-platform/pull/14710/commits/eb11c98ce6d52cb84b3f02b79a0b3e71b9224a63#diff-f08cd4e23bee4cec4156a359aa5e0af7R258), to catch when a `staff` user is requesting grades for a non-existent user.  This is required now because previously, this case was caught when the `get_course_with_access` method was called.